src/react/components/pages/editorPage/canvas.tsx

```diff
@@ -311,17 +352,11 @@ export default class Canvas extends React.Component<ICanvasProps, ICanvasState>
         }
     }
 
-    /**
-     * Add tag to region if not there already, remove tag from region
-     * if already contained in tags. Update tags in CanvasTools editor
-     * @param region Region to add or remove tag
-     * @param tag Tag to add or remove from region
-     */
-    private toggleTagOnRegion = (region: IRegion, tag: string) => {
-        CanvasHelpers.toggleTag(region.tags, tag);
-        this.editor.RM.updateTagsById(region.id, CanvasHelpers.getTagsDescriptor(this.props.project.tags, region));
-        const updatedRegions = this.state.currentAsset.regions.map((r) => (r.id === region.id) ? region : r);
-        const updatedSelectedRegions = this.state.currentAsset.regions.map((r) => (r.id === region.id) ? region : r);
+    private updateRegions = (updates: IRegion[], updatedSelectedRegions: IRegion[]) => {
```

Comment:
What is the different between updates vs updatedSelectedRegions?  Let's add some more docs to make this easier to understand.